daemon/cmd/hubble.go

```diff
@@ -50,21 +80,21 @@ func (d *Daemon) launchHubble() {
 		logger.WithError(err).Error("Failed to initialize Hubble")
 		return
 	}
-	observerServer, err := hubbleServer.NewLocalServer(payloadParser, logger,
+	d.hubbleServer, err = hubbleServer.NewLocalServer(payloadParser, logger,
 		serveroption.WithMaxFlows(option.Config.HubbleFlowBufferSize),
 		serveroption.WithMonitorBuffer(option.Config.HubbleEventQueueSize),
 		serveroption.WithCiliumDaemon(d))
 	if err != nil {
 		logger.WithError(err).Error("Failed to initialize Hubble")
 		return
 	}
-	go observerServer.Start()
-	d.monitorAgent.GetMonitor().RegisterNewListener(d.ctx, hubble.NewHubbleListener(observerServer))
+	go d.hubbleServer.Start()
+	d.monitorAgent.GetMonitor().RegisterNewListener(d.ctx, hubble.NewHubbleListener(d.hubbleServer))
 
 	srv, err := hubbleServe.NewServer(logger,
```

Comment:
This is why I think it's confusing: it looks like you create a new hubble server with a hubble server as an option (the observer service).